infrastructure/ai/providers/openai_provider.py

```
@@ -48,9 +52,25 @@ async def generate(
         try:
-            if self._use_legacy:
-                return await self._generate_via_chat(prompt, config)
-            return await self._generate_via_responses(prompt, config)
+            base_url = self.settings.base_url or "https://api.openai.com/v1"
+            use_responses = not self._use_legacy and base_url not in self.__class__._fallback_to_chat_cache
+
+            if use_responses:
+                try:
+                    return await self._generate_via_responses(prompt, config)
+                except (openai.NotFoundError, openai.BadRequestError, RuntimeError) as e:
+                    logger.info(f"Responses API unsupported for {base_url}, falling back to chat completions: {str(e)}")
```

Ruff (0.15.10)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)